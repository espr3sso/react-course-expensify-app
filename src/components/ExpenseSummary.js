import React from 'react';
import { connect } from 'react-redux';
import getVisibleExpenses from './../selectors/expenses';
import getExpensesTotal from './../selectors/expenses-total';
import numeral from 'numeral';

export const ExpenseSummary = ({ expensesTotal, expensesCount }) => {
  const expenseWord = expensesCount === 1 ? 'expense' : 'expenses';
  return (
    <div>
      <h1>Viewing {expensesCount} {expenseWord} totalling {expensesTotal}</h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const visibleExpenses = getVisibleExpenses(state.expenses, state.filters);
  return {
    expensesTotal: numeral(getExpensesTotal(visibleExpenses) / 100).format('$0,0.00'),
    expensesCount: visibleExpenses.length
  };
};

export default connect(mapStateToProps)(ExpenseSummary);