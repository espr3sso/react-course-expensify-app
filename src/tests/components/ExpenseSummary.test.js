import React from 'react';
import { ExpenseSummary } from './../../components/ExpenseSummary';
import { shallow } from 'enzyme';

test('should render ExpenseSummary correctly with 1 expense', () => {
  const wrapper = shallow(<ExpenseSummary expensesCount={1} expensesTotal={321} />);
  expect(wrapper).toMatchSnapshot();
});

test('should render ExpenseSummary correctly with many expense', () => {
  const wrapper = shallow(<ExpenseSummary expensesCount={22} expensesTotal={321231} />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly sum many expenses', () => {
  
});

test('', () => {
  
});