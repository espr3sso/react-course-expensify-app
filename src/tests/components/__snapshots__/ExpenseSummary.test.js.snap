// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary correctly with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    321
  </h1>
</div>
`;

exports[`should render ExpenseSummary correctly with many expense 1`] = `
<div>
  <h1>
    Viewing 
    22
     
    expenses
     totalling 
    321231
  </h1>
</div>
`;
